tools/topology/topology2/include/common/audio_format.conf

```diff
@@ -0,0 +1,237 @@
+	DefineAttribute."out_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output valid bit depth
+	DefineAttribute."out_valid_bit_depth" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output channel count
+	DefineAttribute."out_channels" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			min 1
+			max 8
+		}
+	}
+
+	# output channel map
+	DefineAttribute."out_ch_map" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output channel config
+	DefineAttribute."out_ch_cfg" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	# output interleaving style
+	DefineAttribute."out_interleaving_style" {
+		type	"string"
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			!valid_values [
+				"interleaved"
+				"non-interleaved"
+			]
+			!tuple_values [
+				0
+				1
+			]
+		}
+	}
+
+	# output format config
+	DefineAttribute."out_fmt_cfg" {
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	#
+	# input buffer size
+	#
+	DefineAttribute."ibs" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	#
+	# output buffer size
+	#
+	DefineAttribute."obs" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	#output sample_type
+	DefineAttribute."out_sample_type" {
+		type string
+		# Token set reference name
+		token_ref	"sof_tkn_cavs_audio_format.word"
+		constraints {
+			!valid_values [
+				$SAMPLE_TYPE_MSB_INTEGER
+				SAMPLE_TYPE_LSB_INTEGER
+				$SAMPLE_TYPE_SIGNED_INTEGER
+				$SAMPLE_TYPE_UNSIGNED_INTEGER
+				$SAMPLE_TYPE_FLOAT
+			]
+		}
+	}
+
+	DefineAttribute."dma_buffer_size" {
+		# Token set reference name and type
+		token_ref	"sof_tkn_cavs_audio_format.word"
+	}
+
+	attributes {
+		!constructor [
+			"instance"
+		]
+
+		#
+		# audio_format objects instantiated within the same alsaconf node must have unique
+		# instance attribute values
+		#
+		unique	"instance"
+	}
+
+	in_rate		48000
+	in_bit_depth		16
+	in_valid_bit_depth	16
+	in_channels		2
+	in_interleaving_style	"interleaved"
+	in_sample_type		$SAMPLE_TYPE_MSB_INTEGER
+	out_rate		48000
+	out_bit_depth		16
+	out_valid_bit_depth	16
+	out_channels		2
+	out_interleaving_style	"interleaved"
+	out_sample_type	$SAMPLE_TYPE_MSB_INTEGER
+	in_ch_cfg	$CHANNEL_CONFIG_STEREO
+	in_ch_map	$CHANNEL_MAP_STEREO
+	out_ch_cfg	$CHANNEL_CONFIG_STEREO
+	out_ch_map	$CHANNEL_MAP_STEREO
+
+	# math expression for computing input/output fmt_cfg
+	in_fmt_cfg	"$[($in_channels | ($in_valid_bit_depth * 256)) | ($in_sample_type * 65536)]"
+	out_fmt_cfg	"$[($out_channels | ($out_valid_bit_depth * 256)) | ($out_sample_type * 65536)]"
+
+	# math expression for computing input/put buffer sizes
+	ibs "$[($in_channels * ($in_rate / 1000)) * ($in_bit_depth / 8)]"
```

Comment:
how to deal with 44.1KHZ ? do we need to support it or not ?

Reply:
one way is that: we allocate 44.1KHZ as size of 48KHZ, only use part of it

Reply:
sorry not getting the comment @RanderWang are you saying the calculation should always use 48KHz even if in_rate is 44.1?

Reply:
yes, we may waste some memory